meson.build

```diff
@@ -95,13 +95,14 @@ default_c_flags = []
 default_cpp_flags = []
 
 add_project_arguments([
-    '-Wl,-z,relro',
-    '-Wl,-z,noexecstack',
-    '-pie',
     '-fstack-protector-strong',
     '-fcf-protection=full',
 ], language: [ 'c', 'cpp' ])
-
+add_project_link_arguments([
+    '-Wl,-z,relro',
+    '-Wl,-z,noexecstack',
+    '-pie',
+], language : [ 'c', 'cpp' ])
```

Comment:
In a second though, we need to clarify this options are only for `native` compilations and not for `cross` compilations to not break Windows build.
```
101 add_project_link_arguments([
102     '-Wl,-z,relro',
103     '-Wl,-z,noexecstack',
104     '-pie',
105 ], language : [ 'c', 'cpp' ],
106 native: true)
```

Reply:
Or, rather, they are for not-windows. They do apply to FreeBSD, for example. They wouldn't apply to macOS.

There must be a way to dynamically check them. Will update.